module-system/sov-modules-macros/src/lib.rs

```diff
@@ -82,6 +84,15 @@ pub fn codec(input: TokenStream) -> TokenStream {
     handle_macro_error(codec_macro.derive_message_codec(input))
 }
 
+/// Specifies that a constant should be configured from the manifest file.
+#[proc_macro_attribute]
```

Comment:
One alternative to consider is to, instead of making it an attribute macro (i.e. `#[config_constant]`), a simple macro, as it might be easier to use. Example:

from:
```rust
#[config_constant]
pub const TEST_U32: u32 = 0;
```

to:
```rust
config_constant!(TEST_U32, u32);
```

Reply:
I think it would be possible to have something like

```rust
#[config_constant]
pub const TEST_U32: u32;
```

And the macro expansion will append the `= 42`. Having it declared with a default value might give the user the wrong impression the constant will fallback to the provided value if the parsing fails (or the constants.json file is absent)

Reply:
TBH, I think the first syntax is a little bit more readable.

Reply:
Wdyt of introducing that default value as fallback to when `constants.json` is absent, or has no `constants.TEST_U32`? We could ofc add this as future functionality by creating an issue

Reply:
Oops, I hadn't refreshed the page, so my comment missed your second one. 

I like this syntax a lot
```rust
#[config_constant]
pub const TEST_U32: u32;
```

I'm afraid that using the default value as a fallback will introduce a footgun where the rollup just silently accepts not being configured , so I think it's preferable to fail explicitly when the constants aren't set.

Reply:
Let's plan to convert to this new syntax in a follow-up. Don't want to block this work, and it will be a bit painful because we can't rely on syn's default parser since the declaration is incomplete.

Reply:
It actually turned out not to be too complicated, so I went ahead!